packer/plugin-getter/github/getter.go

```diff
@@ -0,0 +1,223 @@
+package github
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-github/v33/github"
+	plugingetter "github.com/hashicorp/packer/packer/plugin-getter"
+	"golang.org/x/oauth2"
+)
+
+const (
+	ghTokenAccessor  = "PKR_GITHUB_API_TOKEN"
+	defaultUserAgent = "packer-plugin-getter"
+)
+
+type Getter struct {
+	Client    *github.Client
+	UserAgent string
+}
+
+var _ plugingetter.Getter = &Getter{}
+
+func tranformChecksumStream() func(in io.ReadCloser) (io.ReadCloser, error) {
+	return func(in io.ReadCloser) (io.ReadCloser, error) {
+		defer in.Close()
+		rd := bufio.NewReader(in)
+		buffer := bytes.NewBufferString("[")
+		json := json.NewEncoder(buffer)
+		for i := 0; ; i++ {
+			line, err := rd.ReadString('\n')
+			if err != nil {
+				if err != io.EOF {
+					return nil, fmt.Errorf(
+						"Error reading checksum file: %s", err)
+				}
+				break
+			}
+			parts := strings.Fields(line)
+			switch len(parts) {
+			case 2: // nominal case
+				checksumString, checksumFilename := parts[0], parts[1]
+
+				if i > 0 {
+					_, _ = buffer.WriteString(",")
+				}
+				if err := json.Encode(struct {
+					Checksum string `json:"checksum"`
+					Filename string `json:"filename"`
+				}{
+					Checksum: checksumString,
+					Filename: checksumFilename,
+				}); err != nil {
+					return nil, err
+				}
+			}
+		}
+		_, _ = buffer.WriteString("]")
+		return ioutil.NopCloser(buffer), nil
```

Comment:
Will this return only `"[]"` if is not a nominal case?

Reply:
yes

Reply:
( leaving it like so, I think this is still okay as it is valid JSON )